src/setup.rs

```diff
@@ -18,19 +18,34 @@ const VERSION: &str = env!("CARGO_PKG_VERSION");
 /// Set by our `build.rs`, reflects the Rust target triple we're building for
 const TARGET: &str = env!("TARGET");
 
-/// Where Kani has been installed. Typically `~/.kani/kani-1.x/`
-pub fn kani_dir() -> PathBuf {
-    home::home_dir().expect("Couldn't find home dir?").join(".kani").join(format!("kani-{VERSION}"))
+/// The directory where Kani is installed, either:
+///  * `${KANI_HOME}/kani-<VERSION>` if the environment variable
+///    `KANI_HOME` is set.
+///  * `${HOME}/.kani/kani-<VERSION>` where `HOME` is the canonical definition
+///    of home directory used by Cargo and rustup.
+pub fn kani_dir() -> Result<PathBuf> {
+    let (home_dir, origin) = match env::var("KANI_HOME") {
+        Ok(val) => (PathBuf::from(val), "KANI_HOME"),
+        Err(_) => (home::home_dir().expect("couldn't find home directory").join(".kani"), "HOME"),
+    };
+    if !home_dir.is_dir() {
```

Comment:
Can you please test if this works even when the user is first installing Kani. I.e., delete `~/.kani` and try to run setup.

I'm asking because on line `54` we are explicitly creating `kani_dir` if it doesn't exist.

Reply:
Actually, I think the regression is failing because of that: https://github.com/model-checking/kani/actions/runs/3858944820/jobs/6578010318

Reply:
There's an easy way to fix this (pushing ".kani" after the check), but I think the code & error message are too confusing then. Let me try something else.

Reply:
Added two separate functions to get the custom/default Kani home directory since the two cases are treated differently in terms of error handling. This is hopefully clearer and works.